testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -1,326 +1,325 @@
 TestUGen_RTAlloc : UnitTest {
 
-	var server;
+	classvar server;
 
-	setUp {
-		server = Server(this.class.name);
-		server.options.memSize = 1024;
+	*initClass {
+		passVerbosity = UnitTest.brief;
+	}
+
+	*setUpClass {
+		server = Server(this.name);
+		server.options.sampleRate = 48000;
 		server.options.blockSize = 64;
-		this.bootServer(server);
+		server.options.memSize = 2 ** 13; // scsynth default
+		// - tests fail with memSize < 256 (GVerb fails)
+		// - testing with memSize >= 2 ** 20 would require multiple allocations
+		//   (allocation size itself can overflow size_t for some UGens)
+		server.bootSync;
 	}
 
-	tearDown {
+	*tearDownClass {
 		server.quit.remove;
 	}
 
-	test_allocFail_clearUnit {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(DC.ar(1), 100); }.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
-			cond.signalOne();
-		});
+	// Async helpers
 
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 0.0, "a failed RTAlloc should clear unit outputs")
-	}
-
-	test_allocFail_continueProcessing_sameNode {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(Silent.ar, 100); DC.ar(1)}.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
+	awaitSynthOutput { |synthFn, runtimeSamples=1, timeout=1|
+		var cond = CondVar(), synthOutput = nil;
+		var runtimeSeconds = runtimeSamples / server.sampleRate;
+		// allow extra timeout for longer tests (e.g. testFFT with big memSize)
+		if (runtimeSeconds > timeout) { 
+			timeout = runtimeSeconds + timeout
+		};
+		synthFn.loadToFloatArray(runtimeSeconds, server) { |out|
+			synthOutput = out;
 			cond.signalOne();
-		});
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 1.0, "a failed RTAlloc should not block other units in the same node")
+		};
+		cond.waitFor(timeout) { synthOutput.notNil };
+		^synthOutput;
 	}
 
-	test_allocFail_continueProcessing_otherNodes {
-		var cond = CondVar(), output = -1;
-		server.makeBundle(nil) {
-			{ DelayN.ar(Silent.ar, 100) }.play(server);
-			{ DC.ar(1) }.loadToFloatArray(1 / server.sampleRate, server, {|val|
-				output = val[0];
-				cond.signalOne();
-			});
+	assertAllocPass { |name, func, runtimeSamples|
```

Comment:
To be more explicit about timing, this could perhaps be called `awaitAllocPass`.

Reply:
I see your point, however I would also keep the assert in there....`awaitAssertAllocPass`?

Reply:
I would assume that the assertion is implicit in passing allocation test. What wasn't clear when first seeing the method was that it blocks the thread, so I thought `awaitAllocPass` would explain the function and side effect of the method.
But in any case it's just a suggestion...